lib/ansible/modules/cloud/amazon/ec2_vpc_route_table_info.py

```diff
@@ -52,81 +57,174 @@
+          returned: when vpc peering available
+          type: string
+          sample: vpc-a01106c2
+
+'''
+import json
 
 try:
-    import boto.vpc
-    from boto.exception import BotoServerError
-    HAS_BOTO = True
+    from botocore.exceptions import ClientError, BotoCoreError
 except ImportError:
-    HAS_BOTO = False
-
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import AnsibleAWSError, connect_to_aws, ec2_argument_spec, get_aws_connection_info
-
-
-def get_route_table_info(route_table):
+    pass  # caught by imported HAS_BOTO3
 
-    # Add any routes to array
-    routes = []
-    associations = []
-    for route in route_table.routes:
-        routes.append(route.__dict__)
-    for association in route_table.associations:
-        associations.append(association.__dict__)
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (ec2_argument_spec, ansible_dict_to_boto3_filter_list,
+                                      camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict)
 
-    route_table_info = {'id': route_table.id,
-                        'routes': routes,
-                        'associations': associations,
-                        'tags': route_table.tags,
-                        'vpc_id': route_table.vpc_id
-                        }
 
-    return route_table_info
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
 
 
-def list_ec2_vpc_route_tables(connection, module):
+def list_route_tables(route_table, module):
+    params = dict()
 
-    filters = module.params.get("filters")
-    route_table_dict_array = []
+    params['Filters'] = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    params['RouteTableIds'] = module.params.get('route_table_ids')
 
     try:
-        all_route_tables = connection.get_all_route_tables(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
-
-    for route_table in all_route_tables:
-        route_table_dict_array.append(get_route_table_info(route_table))
-
-    module.exit_json(route_tables=route_table_dict_array)
+        result = json.loads(json.dumps(route_table.describe_route_tables(**params), default=date_handler))
+    except ValueError as e:
+        module.fail_json_aws(e, msg="Cannot validate JSON data")
+    except (ClientError, BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Could not describe route tables")
+    snaked_route_tables = [camel_dict_to_snake_dict(route_table,
+                           ignore_list=['Tags']) for route_table in result['RouteTables']]
+    if snaked_route_tables:
+        for route_table in snaked_route_tables:
+            route_table['tags'] = boto3_tag_list_to_ansible_dict(route_table.get('tags', []))
+    module.exit_json(changed=False, route_tables=snaked_route_tables)
 
 
 def main():
     argument_spec = ec2_argument_spec()
```

Comment:
AnsibleAWSModule automatically adds ec2_argument_spec() so you can simplify 213 through 219

```
    argument_spec = dict(
        route_table_ids=dict(default=[], type='list'),
        filters=dict(default={}, type='dict')
    )